src/Microsoft.TestPlatform.Client/TestPlatform.cs

```diff
@@ -92,6 +92,16 @@ public IDiscoveryRequest CreateDiscoveryRequest(DiscoveryCriteria discoveryCrite
             // Update cache with Extension Folder's files
             this.AddExtensionAssemblies(discoveryCriteria.RunSettings);
 
+            // Update and initialize loggers only when DesignMode is false
+            var runConfiguration = XmlRunSettingsUtilities.GetRunConfigurationNode(discoveryCriteria.RunSettings);
+            if (runConfiguration.DesignMode == false)
+            {
+                this.AddExtensionAssembliesFromSource(discoveryCriteria.Sources);
+
+                // Initialize loggers
+                TestLoggerManager.Instance.InitializeLoggers();
```

Comment:
Now since this happens for both discovery and execution, we should think if there is a better way to do this.

Reply:
Functioning is not exactly same. In case of execution and discovery, we are calling different overriden  methods.